### ISSUE TITLE
Downloaded image has 0 pixels

### ISSUE THREAD

**Issue by rahulbbit** [SDWebImage/SDWebImage#3004]
### New Issue Checklist

* [.] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [.] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [.] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name                          |  ios 
 Platform Version       | iOS 13
 SDWebImage Version     | 5.7.4
 Integration Method     | cocoapods
 Xcode Version           | Xcode 11 
 Repro rate              |  all the time (100%)
 Repro with our demo prj | this does not happen in demo
 Demo project link     | image link "https://s3.ap-south-1.amazonaws.com/hootu-admin/video_image/1583388158.png"

### Issue Description and Steps

I am using SDWebImage (5.7.4) from pods, but i am still facing this issue. It works in demo but not from pods, how do i solve this?

**Comment by dreampiggy** [comment]
> It works in demo but not from pods, how do i solve this?

So this sounds like a issue of your project.

Setup the debugger breakpoint at the decoding part, to see why the image decoding failed.

**Comment by dreampiggy** [comment]
This line: https://github.com/SDWebImage/SDWebImage/blob/5.7.4/SDWebImage/Core/SDImageLoader.m#L82

**Comment by rahulbbit** [comment]
[imageDemo 2.zip](https://github.com/SDWebImage/SDWebImage/files/4597841/imageDemo.2.zip)

This is a fresh demo i just created, and in that it always goes into error.

**Comment by dreampiggy** [comment]
@rahulbbit 

This URL is a webp image, not png. Why this URL end with `.png` ?

You should add WebPCoder to the coders manager.

```swift
import SDWebImage
import SDWebImageWebPCoder

    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        // Override point for customization after application launch.
        SDImageCodersManager.shared.addCoder(SDImageWebPCoder.shared)
        return true
    }
```

**Comment by rahulbbit** [comment]
This worked, thank you so much. Even i was confused that the extension is png then why its causing error

**Comment by dreampiggy** [comment]
> Even i was confused that the extension is png then why its causing error

WebP is not supported by SD's default coders.

So, if you don't add WebPCoder, you decoding part failed.

Decoding failed cause a nil image.

Nil image will show error `Downloaded image has 0 pixels`.

If you think this `Downloaded image has 0 pixels` is confusing, you can fire a feature request, for example:

1. When `image is nil`, the error description should be `Downloaded image decoding failed`
2. When `image not nil, but the image width or height is zero`, the error descriiption should be `Downloaded image has 0 pixels`

**Comment by dreampiggy** [comment]
Does chaning the description of error, can help you for this case ?

**Comment by rahulbbit** [comment]
In my case image was nil and this was the error description 'Downloaded image has 0 pixels'

**Comment by dreampiggy** [comment]
The error description can be polished in #3005.

**Comment by dreampiggy** [comment]
Available in SDWebImage [5.8.0](https://github.com/SDWebImage/SDWebImage/releases/tag/5.8.0)

**Comment by VictorZhang2014** [comment]
I have encountered the problem too, and my solution was done by ADDING `"SD_WEBP=1"` into your `SDWebImage` target -> `Build Settings` -> `Preprocessor Macros`.

<img width="897" alt="WX20210408-213028@2x" src="https://user-images.githubusercontent.com/6853884/114035986-3a05ee80-98b2-11eb-880e-17423c0a460b.png">
